docs/extend/middleware.md

```diff
@@ -1 +1,108 @@
-#  Middleware
\ No newline at end of file
+#  Middleware
+
+Middleware is a nifty way to filter HTTP requests in Flarum. This can allow you to, for example, run a check against an external database checking a user when they register, either allowing them to continue the process or not. The possibilities are endless!
+
+## Adding Middleware In Your Extension
+
+To add a new middleware, add a listener in your `extend.php` file:
+
+```php
+$events->subscribe(Listeners\AddMiddleware::class);
```

Comment:
This will no longer be needed in beta.12 this can be changed to:

```
use Flarum\Extend

return [
    // Add middleware to user facing Forum
    (new Extend\Middleware('forum'))->add(MiddlewareName::class),
    // Add middleware to API
    (new Extend\Middleware('api'))->add(MiddlewareName::class),
    // Add middleware to admin frontend
    (new Extend\Middleware('admin'))->add(MiddlewareName::class),
];
```

Reply:
Yeah, saw that. Great improvement. And you can make the middleware exclusive to specific frontends too?? Very nifty.